XenAdmin/SettingsPanels/CPUMemoryEditPage.cs

```diff
@@ -279,15 +348,33 @@ private bool HasTopologyChanged
             }
         }
 
+        private long SelectedVcpusMax
+        {
+            get
+            {
+                return (long)comboBoxVCPUs.SelectedItem;
+            }
+        }
+
+        private long SelectedVcpusAtStartup
+        {
+            get
+            {
+                return isVcpuHotplugSupported ? (long)comboBoxInitialVCPUs.SelectedItem : (long)comboBoxVCPUs.SelectedItem;
+            }
+        }
+
         public AsyncAction SaveSettings()
         {
             List<AsyncAction> actions = new List<AsyncAction>();
 
-            if (HasVCPUChanged)
+            if (HasVCPUChanged || HasVCPUsAtStartupChanged)
             {
                 vm.VCPUWeight = Convert.ToInt32(_CurrentVCPUWeight);
-                if (_OrigVCPUs != (long)comboBoxVCPUs.SelectedItem)
-                    actions.Add(new ChangeVCPUSettingsAction(vm, (long)comboBoxVCPUs.SelectedItem));
+                if (_OrigVCPUs != (long)comboBoxVCPUs.SelectedItem || HasVCPUsAtStartupChanged)
+                {
+                    actions.Add(new ChangeVCPUSettingsAction(vm, (long)comboBoxVCPUs.SelectedItem, SelectedVcpusAtStartup));

```

Comment:
I think it would be clearer to write SelectedVcpusMax as the second argument, and hide away how we calculate it.